### PR TITLE
Other messages new sending

### PR DESCRIPTION
**Pull request by Bilb**
* updateMedium groups to new sending pipeline
* remove unused function in sendmessage.js
* update retrySend and resend to new pipeline

### CODE REVIEW

File: js/models/conversations.js [hunk 1]
```diff
@@ -1916,26 +1820,32 @@
...
         );
+        message.trigger('pending');
+
+        members.forEach(member => {
+          const memberPubKey = new libsession.Types.PubKey(member);
+          libsession
+            .getMessageQueue()
+            .sendUsingMultiDevice(memberPubKey, mediumGroupCreateMessage);
+        });
+
```
Comment (Bilb) [inline js/models/conversations.js:1848]: this need to use an updated pipeline to be able to encode them with sender keys

File: js/models/messages.js [hunk 2]
```diff
@@ -1031,65 +1031,52 @@
...
+      const chatMessage = new libsession.Messages.Outgoing.ChatMessage({
+        body,
+        timestamp: this.get('sent_at'),
+        attachments,
+        quote: quoteWithData,
+        preview: previewWithData,
+        expireTimer: this.get('expireTimer'),
+      });
       // Special-case the self-send case - we send only a sync message
       if (recipients.length === 1 && recipients[0] === this.OUR_NUMBER) {
```
Comment (Mikunj) [inline js/models/messages.js:1043]: i think this case can also be simplified to
```
if (recipients.length === 1 && recipients[0] === this.OUR_NUMBER) {
    await this.markMessageSyncOnly();
}
```
and then let the case below send it using the queue which automatically handles sync messages.

File: js/models/messages.js [hunk 3]
```diff
@@ -1031,65 +1031,52 @@
...
       }
 
-      return this.send(conversation.wrapSend(promise));
+      this.trigger('pending');
+      // TODO should we handle open groups message here too? and mediumgroups
+      // Not sure there is the concept of retrySend for those
+      const closedGroupChatMessage = new libsession.Messages.Outgoing.ClosedGroupChatMessage(
+        {
+          chatMessage,
+          groupId: this.get('conversationId'),
```
Comment (Mikunj) [inline js/models/messages.js:1067]: are you sure `groupId` is `conversationId`? i thought `conversationId` had a prefix attached to it or something 🤔

File: js/models/messages.js [hunk 4]
```diff
@@ -1125,38 +1111,43 @@
 
       const quoteWithData = await loadQuoteData(this.get('quote'));
       const previewWithData = await loadPreviewData(this.get('preview'));
+      const chatMessage = new libsession.Messages.Outgoing.ChatMessage({
+        body,
+        timestamp: this.get('sent_at'),
+        attachments,
```
Comment (Mikunj) [inline js/models/messages.js:1117]: Same as the similar comment above saying attachments are not in the correct format here

File: js/models/messages.js [hunk 5]
```diff
@@ -1125,38 +1111,43 @@
...
-          quoteWithData,
-          previewWithData,
-          this.get('sent_at'),
-          this.get('expireTimer'),
-          profileKey
-        );
-        return this.sendSyncMessageOnly(dataMessage);
+        this.trigger('pending');
+        await this.markMessageSyncOnly();
+        const syncMessage = libsession.Utils.SyncMessageUtils.from(chatMessage);
```
Comment (Mikunj) [inline js/models/messages.js:1127]: we can let the queue handle sync messages so we can fallback to the case below

File: js/models/conversations.js [hunk 6]
```diff
@@ -1304,19 +1304,25 @@
...
-          const dataMessage = await textsecure.messaging.getMessageProto(
-            destination,
-            messageBody,
-            finalAttachments,
-            quote,
-            preview,
-            now,
-            expireTimer,
-            profileKey
+          await message.markMessageSyncOnly();
```
[the remark this answers is not in the document]
Comment (Bilb) [reply js/models/conversations.js:1321]: Updated all `getMessageQueue().sendSyncMessageOnly` to use `sendMultiDevice()`

File: js/models/conversations.js [hunk 7]
```diff
@@ -1364,51 +1370,30 @@
...
-            // numbers = [destination];
-            // options.isMediumGroup = true;
-            throw new Error('To implement');
+            const mediumGroupChatMessage = new libsession.Messages.Outgoing.MediumGroupChatMessage(
+              {
+                chatMessage,
+                groupId: destination,
+              }
+            );
+            const members = this.get('members');
```
[the remark this answers is not in the document]
Comment (Bilb) [reply js/models/conversations.js:1388]: It's not, we will need to add the feature to the pipeline but tslint was complaining when variable was not in use.

File: js/models/messages.js (hunk 2 above)
Comment (Bilb) [reply js/models/messages.js:1043]: updated

File: js/models/messages.js (hunk 3 above)
Comment (Bilb) [reply js/models/messages.js:1067]: It's already what was used before: https://github.com/loki-project/session-desktop/pull/1197/files/1650d758a54bf27236ab4570defd9ceefdbb6871#diff-fdb432d7888e10d18d02c7becd34c5e4L1084

File: js/models/messages.js (hunk 5 above)
Comment (Bilb) [reply js/models/messages.js:1127]: fixed

File: libtextsecure/sendmessage.js [hunk 8]
```diff
@@ -379,8 +379,14 @@ MessageSender.prototype = {
     const syncMessages = await Promise.all(
       chunked.map(c => libloki.api.createContactSyncMessage(c))
     );
+    const pubKey = textsecure.storage.user.getNumber();
+
+    const currentPubKey = new libsession.Types.PubKey(pubKey);
+
     const syncPromises = syncMessages.map(syncMessage =>
-      libsession.getMessageQueue().sendSyncMessage(syncMessage)
+      libsession
```
Comment (Mikunj) [inline libtextsecure/sendmessage.js:387]: The previous change was fine as you're sending a proper sync message.
```
libsession.getMessageQueue().sendSyncMessage(syncMessage)
```

I think you got confused with what we were doing in `conversation.js`. The reason we didn't call `sendSyncMessage` there for stuff like expiration update is that the expiration message was being sent to the user (doesn't matter if it's the current user or not) anyway in the lines below.

Here we want to send a sync message to ourselves and not any other user.

**Pull request merged**